astropy/coordinates/angles.py

```diff
@@ -660,7 +674,14 @@ def wrap_angle(self):
 
     @wrap_angle.setter
     def wrap_angle(self, value):
-        self._wrap_angle = Angle(value)
+        if not isinstance(value, Angle):
```

Comment:
In 322a57d I removed this test, and instead added some speedups to `Angle(Angle(...))`.  @mhvk, you might want to double check that the assumptions I made there are correct because I'm not 100% positive.  My thinking is that a lot of validation can be skipped if we're already looking at a valid Angle object (and has hence already been through the `Angle.__new__` gauntlet once before).